testrepository/tests/commands/test_run.py

```diff
@@ -539,3 +622,90 @@ def test_returncode_nonzero_fail_appended_to_content(self):
             expected_content = _b('foo\nbar\ntest: process-returncode\n'
                 'failure: process-returncode [\n returncode 1\n]\n')
         self.assertEqual(expected_content, content)
+
+
+class TestLoadingIds(BaseTestCommand):
+
+    scenarios = [
+        ('noprofile', {'profile': ''}),
+        ('profile', {'profile': 'DEFAULT'})]
+
+    def test_load_list_passes_ids(self):
+        list_file = tempfile.NamedTemporaryFile()
+        self.addCleanup(list_file.close)
+        expected_ids = set(['foo', 'quux', 'bar'])
+        load_ids = set(expected_ids)
+        if self.profile:
+            load_ids = apply_profiles([self.profile], load_ids)
```

Comment:
Conditionals in tests aren't great. Perhaps the profile scenarios should be objects?